adservertargeting/respdataprocessor.go

```diff
@@ -82,13 +82,14 @@ func buildBidExt(targetingData map[string]string,
 	}
 	bidExtTargeting, err := jsonutil.Marshal(bidExtTargetingData)
 	if err != nil {
-		warnings = append(warnings, createWarning(err.Error()))
+		// TODO: figure out the intended usage of warnings, here and several lines below
+		// warnings = append(warnings, createWarning(err.Error()))
 		return nil
 	}
 
 	newExt, err := jsonpatch.MergePatch(bid.Ext, bidExtTargeting)
 	if err != nil {
-		warnings = append(warnings, createWarning(err.Error()))
+		// warnings = append(warnings, createWarning(err.Error()))
```

Comment:
Please resolve these TODOs.

Reply:
removed `TODO`s and added the `//nolint: ineffassign,staticcheck` comment.  But it seems to me somewhere between fragile and wrong to append to slice passed by value - is it guaranteed that the caller will see those additions? https://go.dev/play/p/DmZumKNLFYC suggests it won't (or may not - depending on the capacity of the passed slice).  Should we change `buildBidExt` to take a pointer-to-slice `warnings *[]openrtb_ext.ExtBidderMessage` - or even return the `warnings` together with the `json.RawMessage`?